packages/mui-joy/src/AccordionSummary/accordionSummaryClasses.ts

```diff
@@ -0,0 +1,27 @@
+import { generateUtilityClass, generateUtilityClasses } from '../className';
+
+export interface AccordionSummaryClasses {
+  /** Class name applied to the root element. */
+  root: string;
+  /** Class name applied to the button element. */
+  button: string;
+  /** Class name applied to the indicator element. */
+  indicator: string;
+  /** Class name applied when the accordion is disabled. */
+  disabled: string;
+  /** Class name applied when the accordion is expanded. */
+  expanded: string;
```

Comment:
For styling purposes, without these classes, you will have to theme the `AccordionSummary` like this:

```js
extendTheme({
  components: {
    JoyAccordion: {
      styleOverrides: {
        root: {
          [`.${accordionClasses.root}.${accordionClasses.expanded} &`]: {
             …styles
          }
        }
      }
    }
  }
})
```

whereas 
```js
extendTheme({
  components: {
    JoyAccordion: {
      styleOverrides: {
        root: {
          [`&.${accordionSummaryClasses.expanded}`]: {
             …styles
          }
        }
      }
    }
  }
})
```

Reply:
I agree it's easier, but if we do it for the `AccordionSummary` component, shouldn't we do it for the other accordion components as well?

- `AccordionDetails` has `expanded` but not `disabled`
- `AccordionGroup` has neither

We should either add the classes to all or none. Otherwise, it might be confusing.

I would only have these on the root, I think the following is not a bad API:

```js
extendTheme({
  components: {
    JoyAccordion: {
      styleOverrides: {
        root: {
          [`.${accordionClasses.expanded} &.${accordionSummaryClasses.root}`]: {
             …styles
          }
        }
      }
    }
  }
})
```